components/automate-ui/src/app/pages/service-groups/service-groups.component.ts

```diff
@@ -246,21 +246,20 @@ export class ServiceGroupsComponent implements OnInit, OnDestroy {
     // are filtered out via those status filters we have a special state that
     // says, e.g., "none of the services returned warning" that we want to
     // show.
-    this.store.select(serviceGroupsStatus).subscribe((sgStatus) => {
-      if (sgStatus === 'loadingSuccess') {
-        this.store.select(selectedServiceGroupStatus).subscribe((sSgStatus) => {
-          if (sSgStatus === 'loadingSuccess') {
-            this.serviceGroupsList$.subscribe((serviceGroups) => {
-              if (serviceGroups.length > 0) {
-                this.store.select(selectedServiceGroupHealth).subscribe((sgHealth) => {
-                  if (sgHealth.total === 0) {
-                    this.onServiceGroupSelect(null, serviceGroups[0].id);
-                  }
-                });
+    combineLatest([
+      this.store.select(selectedServiceGroupStatus),
+      this.store.select(serviceGroupsStatus),
+      this.store.select(serviceGroupsList),
+      this.store.select(selectedServiceGroupHealth)
+    ]).pipe(
+      takeUntil(this.isDestroyed)
+    ).subscribe(([sSgStatus, sgStatus, serviceGroups, sgHealth]) => {
+      if (sgStatus === 'loadingSuccess' && sSgStatus === 'loadingSuccess') {
+          if (serviceGroups.length > 0) {
+              if (sgHealth.total === 0) {
+                this.onServiceGroupSelect(null, serviceGroups[0].id);
               }
-            });
           }
-        });
       }
     });
```

Comment:
With the `combineLatest` clearing up the nested subscription problem, the code is sufficient, but we can clean it up a bit further. Using the convention of applying filters instead of using conditionals, here we can substantially reduce the cyclomatic complexity by eliminating all the remaining nested conditionals.
```
      filter(([sSgStatus, sgStatus, serviceGroups, sgHealth]) =>
        sgStatus === EntityStatus.loadingSuccess
        && sSgStatus === EntityStatus.loadingSuccess
        && serviceGroups.length > 0
        && sgHealth.total === 0
      ),
      takeUntil(this.isDestroyed)
    ).subscribe(([_sSgStatus, _sgStatus, serviceGroups, _sgHealth]) =>
      this.onServiceGroupSelect(null, serviceGroups[0].id));
```

Reply:
@msorens speaking of conventions, I've noticed a mixture of `this.isDestroyed$` and `this.isDestroyed`. I'm not sure what we'd want to prefer using (other than `isDestroyed$` seems more prevalent I think?)

Should `$` be appended to only `Observable` types, or appended to any type that can be subscribed to? (`Observable`, `Subject`, `ReplaySubject`, etc). I lean towards the latter at the moment

Reply:
Great question, @scottopherson ! I have been kicking that can down the road because I thought it was just for Observables but was not really sure... A quick web search just now revealed people say Observable in one breath but then usually go on to add "... and streams". So I am happy to oblige and go with your suggestion of any type that can be subscribed to. 👍

Reply:
@msorens ah yea, I'm seeing the "and streams" as well.
Yeah to me seems like `$` is useful for immediately identifying what can be done with a certain value (`subscribe`, `useWith$ | async`, use within `pipe` chain) and possible pitfalls that can occur with said value (memory leaks). Though seeing the types for these values in our IDEs helps with that as well 🤷